src/main/scala/mesosphere/marathon/core/readiness/impl/ReadinessCheckExecutorImpl.scala

```diff
@@ -94,9 +98,38 @@ private[readiness] class ReadinessCheckExecutorImpl(implicit actorSystem: ActorS
       ReadinessCheckResult.forSpecAndResponse(check, response).copy(ready = false)
   }
 
+  private val disabledSslContext: SSLContext = {
+    object BlindFaithX509TrustManager extends X509TrustManager {
+      def checkClientTrusted(chain: Array[X509Certificate], authType: String): Unit = {}
+      def checkServerTrusted(chain: Array[X509Certificate], authType: String): Unit = {}
+      def getAcceptedIssuers: Array[X509Certificate] = Array[X509Certificate]()
+    }
+
+    val context = SSLContext.getInstance("TLS")
+    context.init(Array[KeyManager](), Array(BlindFaithX509TrustManager), null)
+    context
+  }
+
+  private val disabledSslConfig = AkkaSSLConfig().mapSettings(s => s.withLoose {
+    s.loose.withAcceptAnyCertificate(true)
+      .withAllowLegacyHelloMessages(Some(true))
+      .withAllowUnsafeRenegotiation(Some(true))
+      .withAllowWeakCiphers(true)
+      .withAllowWeakProtocols(true)
+      .withDisableHostnameVerification(true)
+      .withDisableSNI(true)
+  })
+
   private[impl] def akkaHttpGet(check: ReadinessCheckSpec): Future[AkkaHttpResponse] = {
-    Timeout(check.timeout)(Http().singleRequest(
-      request = RequestBuilding.Get(check.url)
-    ))
+    if (check.url.startsWith("https")) {
+      Timeout(check.timeout)(Http().singleRequest(
```

Comment:
is it an error to set the connection context if https is not used?

Reply:
(answer seems to be no; in which case do we need this if statement?)

Reply:
hmm maybe no, I was not sure, because this is what we do for health checks too :)